analysis/bin/doubleHiggsAnalyser.cc

```diff
@@ -82,135 +113,238 @@ void doubleHiggsAnalyser::Initiate(TString output_file_name) {
-    from2 = abs(from2);
-    if (from2==doubleHiggsAnalyser::Higgs_PID) fromHiggs++;
-    if (from2==doubleHiggsAnalyser::Top_PID) fromTop++;
-    if (from2==doubleHiggsAnalyser::Z_PID) fromZ++;
-    muon2.SetPtEtaPhiM(mu2->PT,mu2->Eta,mu2->Phi,mu2->Mass);
-    
-    lepton1_pt = mu1->PT;
-    lepton2_pt = mu2->PT;
-    missing_et = m->MET;
-    auto muons = muon1+muon2;
-    if (muons.M() > 65 || muon1.DeltaR(muon2) > 1) {
-        return false;
-    }
-    mt = muons.Mt();
+    auto bot1 = static_cast<const Jet *>(jets->At(bottom_iter->second));
+    bottom1.SetPtEtaPhiM(bot1->PT,bot1->Eta,bot1->Phi,bot1->Mass);
+    //int* bottom1_pdg = isFrom(particles, bottom_iter->second); // bottom truth matching
+    //bot1_mother = abs(bottom1_pdg[0]);
+    //bot1_grmother = abs(bottom1_pdg[1]);
+    bottom_iter ++;
+    auto bot2 = static_cast<const Jet *>(jets->At(bottom_iter->second));
+    bottom2.SetPtEtaPhiM(bot2->PT,bot2->Eta,bot2->Phi,bot2->Mass);
+    bottombottom = bottom1+bottom2;
+    //int* bottom2_pdg = isFrom(particles, bottom_iter->second); // bottom truth matching
+    //bot2_mother = abs(bottom2_pdg[0]);
+    //bot2_grmother = abs(bottom2_pdg[1]);
+    // bottom kinematic variables
+    bottom_mass = bottombottom.M();
+    bottom_pt = bottombottom.Pt();
+    bottom_px = bottombottom.Px();
+    bottom_py = bottombottom.Py();
+    bottom_deltaR = fabs(bottom1.DeltaR(bottom2));
+    bottom1_mass = bottom1.M();
+    bottom2_mass = bottom2.M();
+    bottom1_pt = bottom1.Pt();
+    bottom2_pt = bottom2.Pt();
+
+    // invariant mass of bbll
+    bbll = bottombottom + leptonlepton;
+    bbll_mass = bbll.M();
+
+    // missing_et baseline selection
+    if (missing_et<20) {return true;} step++; // base cut 1
+    // lepton baseline selections
+    if (lepton_deltaR > 1) {return true;} step++; // base cut 2
+    if (lepton_mass > 65) {return true;} step++; // base cut 3
+    // bottom baseline selections
+    if (bottom_deltaR > 1.3) {return true;} step++; // base cut 4
+    if (bottom_mass < 95 || bottom_mass > 140) {return true;} step++; // base cut 5
 
-    // get MT2
-    Mt2::LorentzTransverseVector vis_A(Mt2::TwoVector(muon1.Px(), muon1.Py()), muon1.M());
-    Mt2::LorentzTransverseVector vis_B(Mt2::TwoVector(muon2.Px(),muon2.Py()), muon2.M());
+    // MT2
+    Mt2::LorentzTransverseVector vis_A(Mt2::TwoVector(leptonlepton.Px(), leptonlepton.Py()), leptonlepton.M());
+    Mt2::LorentzTransverseVector vis_B(Mt2::TwoVector(bottombottom.Px(), bottombottom.Py()), bottombottom.M());
     Mt2::TwoVector pT_Miss(missing.Px(), missing.Py());
     
     lester_MT2 = asymm_mt2_lester_bisect::get_mT2( // the simpliest way to calculate MT2
-              muon1.M(),muon1.Px(),muon1.Py(),
-              muon2.M(),muon2.Px(),muon2.Px(),
+              leptonlepton.M(),leptonlepton.Px(),leptonlepton.Py(),
+              bottombottom.M(),bottombottom.Px(),bottombottom.Px(),
               missing.Px(),missing.Py(),
               missing.M(),missing.M());
```

Comment:
why does this function take the same input, missing.M(), twice?

Reply:
the 'get_mT2' function in lester bisect header file looks like this and there is only one invisible mass... so... should I set it as 0 instead?

```
static double get_mT2( // returns asymmetric mT2 (which is >=0), or returns a negative number (such as MT2_ERROR) in the case of an error.
    const double mVis1, const double pxVis1, const double pyVis1,
    const double mVis2, const double pxVis2, const double pyVis2,
    const double pxMiss, const double pyMiss,
    const double mInvis1, const double mInvis2,
    const double desiredPrecisionOnMT2=0, // This must be non-negative.  If set to zero (default) MT2 will be calculated to the highest precision available on the machine (or as close to that as the algorithm permits).  If set to a positive value, MT2 (note that is MT2, not its square) will be calculated to within +- desiredPrecisionOnMT2. Note that by requesting precision of +- 0.01 GeV on an MT2 value of 100 GeV can result in speedups of a factor of ...
    const bool useDeciSectionsInitially=true // If true, interval is cut at the 10% point until first acceptance, which gives factor 3 increase in speed calculating kinematic min, but 3% slowdown for events in the bulk.  Is on (true) by default, but can be turned off by setting to false.
  )
```